### PR TITLE
Fix PreFAST errors in dev12 release builds.

### LINKED ISSUE #553: [Test failure] [Daily] [dev12] Legacy build PreFAST errors
Branched from issue #528 

http://dotnet-ci.cloudapp.net/job/Microsoft_ChakraCore/job/daily_dev12_x64_release/51/consoleFull#1026559514ac1851af-571d-472f-b664-6657b078a5dd

```
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.cpp(254): warning C6102: Using 'bcBuffer' from failed function call at line '179'. [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.cpp(338): warning C6102: Using 'bcBuffer' from failed function call at line '319'. [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\chakrartinterface.cpp(28): warning C6102: Using 'modulename' from failed function call at line '27'. [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(52): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'ftell'. See line 51 for an earlier location where this can occur [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(51): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fseek'.  [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(64): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fread'. See line 51 for an earlier location where this can occur [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(65): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fclose'. See line 51 for an earlier location where this can occur [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 [daily_dev12_x---5f770fb3] $ powershell.exe -NonInteractive -ExecutionPolicy ByPass "& 'C:\Users\DOTNET~1\AppData\Local\Temp\2\hudson7084396863305529629.ps1'"
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(52): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'ftell'. See line 51 for an earlier location where this can occur
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(51): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fseek'. 
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(64): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fread'. See line 51 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(65): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fclose'. See line 51 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\common\memory\arenaallocator.cpp(1196): warning C28182: Dereferencing NULL pointer. 'cache' contains the same NULL value as 'curWeakRef' did. See line 1196 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\common\memory\arenaallocator.cpp(1276): warning C28182: Dereferencing NULL pointer. 'cache' contains the same NULL value as 'curWeakRef' did. See line 1276 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\common\memory\recyclerweakreference.h(233): warning C28182: Dereferencing NULL pointer. 'buckets' contains the same NULL value as 'current' did. 
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\jsrt\jsrt.cpp(863): warning C28196: The requirement that '*_Param_(2)!=0' is not satisfied. (The expression does not evaluate to true.)
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\runtime\bytecode\bytecodegenerator.cpp(3733): warning C6011: Dereferencing NULL pointer 'invertedStmt'. 
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\runtime\language\dynamicprofileinfo.cpp(586): warning C6011: Dereferencing NULL pointer 'callSiteInfo[callSiteId].u.polymorphicCallSiteInfo'. See line 578 for an earlier location where this can occur
```

### PR DESCRIPTION
**Pull request by dilijev**
Fixes #553 

This change is [Review on Reviewable](https://reviewable.io/reviews/microsoft/chakracore/551)

### CODE REVIEW

**Comment by dilijev** [comment]
@dotnet-bot
test daily_dev12_x86_release
test daily_dev12_x64_release

**Comment by dilijev** [comment]
@dotnet-bot 
test Windows 7 daily_dev12_x86_release please
test Windows 7 daily_dev12_x64_release please

**Comment by dilijev** [comment]
Tagging people who worked on these files especially in those vicinities:
@jianchun @obastemur 

Please review?

**Comment by dilijev** [comment]
@dotnet-bot 
test Windows 7 daily_dev12_x86_release please
test Windows 7 daily_dev12_x64_release please

File: lib/Jsrt/Jsrt.cpp [hunk 1]
```diff
@@ -858,8 +858,10 @@ STDAPI_(JsErrorCode) JsPointerToString(_In_reads_(stringLength) const wchar_t *s
 }
 
 // TODO: The annotation of stringPtr is wrong.  Need to fix definition in chakrart.h
-// The warning is '*stringPtr' could be '0' : this does not adhere to the specification for the function 'JsStringToPointer'.
+// Warning 6387 is: "'*stringPtr' could be '0' : this does not adhere to the specification for the function 'JsStringToPointer'."
+// Warning 28196 is: "The requirement that '*_Param_(2)!=0' is not satisfied. (The expression does not evaluate to true.)"
 #pragma warning(suppress:6387)
+#pragma warning(suppress:28196)
 STDAPI_(JsErrorCode) JsStringToPointer(_In_ JsValueRef stringValue, _Outptr_result_buffer_(*stringLength) const wchar_t **stringPtr, _Out_ size_t *stringLength)
```
Comment (dilijev) [inline lib/Jsrt/Jsrt.cpp]: @curtisman some archaeology shows that you introduced the existing warning and comment here. I think the warning I disabled is related. Can you confirm? Should we fix this properly now or table it again?

**Comment by dilijev** [comment]
@dotnet-bot 
test Windows 7 daily_dev12_x86_release please
test Windows 7 daily_dev12_x64_release please

**Comment by dilijev** [comment]
@dotnet-bot 
test Windows 7 daily_dev12_x86_debug please
test Windows 7 daily_dev12_x64_debug please
test Windows 7 daily_dev12_x86_test please
test Windows 7 daily_dev12_x64_test please

Since the other flavors should be fixed now.

**Comment by dilijev** [comment]
@dotnet-bot 
test Windows 7 daily_dev12_x86_release please
test Windows 7 daily_dev12_x64_release please

**Comment by dilijev** [comment]
ping - review please? @curtisman @jianchun @digitalinfinity @ianwjhalliday

**Comment by dilijev** [comment]
Link to a comment thread in case I rebase later: https://github.com/dilijev/ChakraCore/commit/8cad615c3587a9291b845b3753ecd9885d11482f#commitcomment-16666693

**Comment by curtisman** [comment]
I wonder whether we should just disable prefast on dev12 build given it's higher rate of false positive?
The Dev14 build should cover most cases anyways?

**Comment by dilijev** [comment]
@curtisman Fair thought. Do you think I should toss this PR or merge it and disable Dev12 PreFAST if it turns up more false positives later?

**Comment by dilijev** [comment]
@dotnet-bot test dev12 tests please

**Comment by dilijev** [comment]
@dotnet-bot test dev12 tests please

**Comment by dilijev** [comment]
I'm closing this PR as the changes are mostly frivolous and might result in incorrect logic. #609 will resolve the issue by changing Jenkins to only run code analysis on configurations where it will give us real information without false positives.

**Pull request closed without merging**